IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -105,19 +109,84 @@ + (MSIDRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestConte
     return info;
 }
 
-+ (SecIdentityRef)copyWPJIdentity:(__unused id<MSIDRequestContext>)context
++ (SecIdentityRef)copyWPJIdentityWithTenantId:(NSString *)tenantId
+                                      context:(id<MSIDRequestContext>)context
+                            certificateIssuer:(NSString **)issuer
+                               privateKeyDict:(NSDictionary **)keyDict
+{
+    NSString *teamId = [[MSIDKeychainUtil sharedInstance] teamId];
+    
+    if (!teamId)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Encountered an error when reading teamID from keychain.");
+        return nil;
+    }
+    
+    NSString *legacySharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin", teamId];
+    
+    SecIdentityRef legacyIdentity = [self copyWPJIdentity:context tag:nil sharedAccessGroup:legacySharedAccessGroup certificateIssuer:issuer privateKeyDict:keyDict];
+    
+    if (legacyIdentity)
+    {
+        if ([NSString msidIsStringNilOrBlank:tenantId])
+        {
+            // ESTS didn't request a specific tenant, just return default one
+            return legacyIdentity;
+        }
+        
+        // Read tenantId for legacy identity
+        NSString *registrationTenantId = [MSIDWorkPlaceJoinUtil getWPJStringDataForIdentifier:kMSIDTenantKeyIdentifier context:context error:nil];
+        
+        // There's no tenantId on the registration, or it mismatches what server requested, keep looking for a better match. Otherwise, return the identity already.
+        if (![NSString msidIsStringNilOrBlank:registrationTenantId]
+            && [registrationTenantId isEqualToString:tenantId])
+        {
+            return legacyIdentity;
+        }
+    }
+    
+    NSString *defaultSharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin.v2", teamId];
+    NSString *tag = [NSString stringWithFormat:@"%@#%@", kWPJPrivateKeyIdentifier, tenantId];
```

Comment:
This is the format for RSA key, the format for ECC key is "%@ ( %@ )-EC"

Reply:
Needs to be addressed in a separate PR once ECC is ready for E2E testing